boards/raspberrypi/rpi_pico_2/Kconfig.defconfig

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2024 Andrew Featherstone
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_RPI_PICO_2
```

Comment:
I think it would be better to use pico2 as the identifier rather than pico_2.
It is use in pico-sdk.

https://github.com/raspberrypi/pico-sdk/blob/efe2103f9b28458a1615ff096054479743ade236/src/boards/include/boards/pico2.h#L20

Reply:
Is this a personal preference or a coding standard/style thing for Zephyr? I prefer the consistent use of underscores as a space separator. What do other reviewers think?

Reply:
Although there is general freedom in this, I think it is customary to follow the "correct name" (trademark, customary notation in vendor-provided documents) defined by the vendor as long as the coding rules and language constraints allow.
In this case, there is a notation used on the pico-sdk, so I think that you should follow that unless it falls under any particular coding rule constraints.
(I think this is an area where Zephyr developers have no freedom.)

Reply:
@yonsch 
I'd love to hear your opinions.

https://github.com/zephyrproject-rtos/zephyr/pull/77368#discussion_r1727805051

Reply:
> I think it would be better to use pico2 as the identifier rather than pico_2. It is use in pico-sdk.

Yes that sounds good.
The board naming was also discussed in the pico-sdk recently:
https://github.com/raspberrypi/pico-sdk/pull/1816#issuecomment-2293167259